File: src/main/java/MusicCube/entities/City.java
package MusicCube.entities;

import javax.persistence.*;

@Entity
public class City {
    @Id
    @GeneratedValue
    private int id;
    @Column(unique = true)
    private String cityName;
    @ManyToOne
    private Country country;

    public City() {};

    public City(String cityName, Country country) {
        this.cityName = cityName;
        this.country = country;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getCityName() {
        return cityName;
    }

    public void setCityName(String cityName) {
        this.cityName = cityName;
    }

    public Country getCountry() {
        return country;
    }

    public void setCountry(Country country) {
        this.country = country;
    }
}

File: Frontend/src/app/Class/City.ts
import {Country} from './Country';

export class City {
  id: number;
  cityName: string;
  country: Country;

  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.cityName = (obj && obj.cityName) || '';
    this.country = (obj && obj.country) || '';
  }

  // Getters

  getId(): number {
    return this.id;
  }

  getCityName(): string {
    return this.cityName;
  }

  getCountry(): Country {
    return this.country;
  }

  // Setters

  setCityName(cityName: string) {
    this.cityName = cityName;
  }

  setCountry(country: Country) {
    this.country = country;
  }

}
